railgun/manager/run.go

```diff
@@ -283,9 +274,36 @@ func Run(ctx context.Context, c *config.Config) error {
 		},
 	}
 
-	if ctrlutils.KnativeCRDExist(mgr.GetClient()) == true {
-		setupLog.Info("ingresses.networking.internal.knative.dev crd already deployed. enable knative contorller.")
-		c.KnativeIngressEnabled = util.EnablementStatusEnabled
+	kongClusterPluginGVR := schema.GroupVersionResource{
+		Group:    configurationv1beta1.SchemeGroupVersion.Group,
+		Version:  configurationv1beta1.SchemeGroupVersion.Version,
+		Resource: "kongclusterplugins",
+	}
+	if ctrlutils.CRDExists(mgr.GetClient(), kongClusterPluginGVR) == true {
+		setupLog.Info("kongclusterplugins.configuration.konghq.com v1beta1 CRD available on cluster.")
+		controller := ControllerDef{
+			IsEnabled: &c.KongClusterPluginEnabled,
+			Controller: &kongctrl.KongV1KongClusterPluginReconciler{
+				Client:           mgr.GetClient(),
+				Log:              ctrl.Log.WithName("controllers").WithName("KongClusterPlugin"),
+				Scheme:           mgr.GetScheme(),
+				Proxy:            prx,
+				IngressClassName: c.IngressClassName,
+			},
+		}
+		controllers = append(controllers, controller)
+	} else {
+		setupLog.Info(`kongclusterplugins.configuration.konghq.com v1beta1 CRD not available on cluster.
+		Disabling KongClusterPlugin controller`)
```

Comment:
If I understand correctly, in 2.0, KongClusterPlugin is disabled when CRD does not exist at first place.

1.3.X mechanism varies, clusterplugin CR sync block until CRD get deployed, and the controller will be enabled after that.